src/App.tsx

```diff
@@ -1,27 +1,116 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classnames from 'classnames';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import 'bulma';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goodsAll, setGoodsAll] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const [loadingAll, setLoadingAll] = useState(false);
+  const [loadingFiveGoods, setLoadingFiveGoods] = useState(false);
+  const [loadingRedGoods, setLoadingRedGoods] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleLoadAllGoods = async () => {
+    setLoadingAll(true);
+    const listOfTodos = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setGoodsAll(listOfTodos);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    setLoadingAll(false);
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleFiveGoods = async () => {
+    setLoadingFiveGoods(true);
+    const listOfTodos = await get5First();
+
+    setGoodsAll(listOfTodos);
+
+    setLoadingFiveGoods(false);
+  };
+
+  const handleRedGoods = async () => {
+    setLoadingRedGoods(true);
```

Comment:
```suggestion
    setLoading(YourEnum.Red);
```
with the enum explained above